include/swift/AST/GenericSignatureBuilder.h

```diff
@@ -314,6 +314,22 @@ class GenericSignatureBuilder {
     /// \c ConstraintResult::Unresolved so the caller knows what happened.
     GenerateUnresolved = 1,
   };
+	
+  /// Describes a struct that contains information about the constraint
+  /// that didn't conform to a protocol or a class
+  struct InvalidConstraint {
+    Type subjectType;
+    Type constraintType;
+    const RequirementSource *reqSource;
+
+    InvalidConstraint(const Type type1, const Type type2,
+                      const RequirementSource *source)
+        : subjectType(type1), constraintType(type2), reqSource(source) {}
+  };
+  
+  /// The set of constraints that are invalid because the constraint
+  /// type isn't constrained to a protocol or a class
+  std::vector<InvalidConstraint> invalidConstraints;
```

Comment:
`invalidConstraints` currently holds constraints of the form `A: B` where we learned that `B` is not a class, protocol, or composition thereof.  Could we rename this to better indicate what it holds (e.g., `invalidIsaConstraints`), or is this (and `InvalidConstraint`) meant to encompass other kinds of failed constraints in the future?